mediagrains/grain.py

```diff
@@ -67,6 +74,17 @@
 __all__ = ["GRAIN", "VIDEOGRAIN", "AUDIOGRAIN", "CODEDVIDEOGRAIN", "CODEDAUDIOGRAIN", "EVENTGRAIN", "attributes_for_grain_type"]
 
 
+def _stringify_timestamp_input(value: Union[SupportsMediaTimestamp, SupportsMediaTimeOffset, str]) -> str:
+    if isinstance(value, SupportsMediaTimestamp):
+        value = mediatimestamp(value).to_sec_nsec()
+    elif isinstance(value, SupportsMediaTimeOffset):
+        value = mediatimeoffset(value).to_sec_nsec()
+    else:
+        assert(isinstance(value, str))
```

Comment:
Given that the value is provided by the client of the library to the `origin_timestamp` setter rather than internal code, shouldn't this raise an error instead?